scripts/data_utils.py

```diff
@@ -436,21 +436,30 @@ def merge_chunks_serially(chunked_content_list: List[str], num_tokens: int) -> G
         yield current_chunk, total_size
 
 
-def get_embedding(text, azure_credential, embedding_endpoint):
+def get_embedding(text, embedding_endpoint, embedding_api_key):

```

Comment:
Hi @shpengmsft can we make it possible to use either azure_credential or API key? The AAD path supports the azd index creation flow. But API key is needed for faster processing on large sets. FYI @pamelafox

Reply:
Hm, why does API key lead to faster processing? Less token refreshes? FYI that we got faster processing by batching up the embeddings- 
https://github.com/Azure-Samples/azure-search-openai-demo/pull/556